CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+### .vscode
+
+The [`.vscode`](/.vscode) directory contains files and configuration specific to the [vscode](https://code.visualstudio.com/docs/getstarted/settings) text editor. These include, settings, extensions and [snippet definitions](#snippets).
+
+### Delivery Tooling
+
+The [`delivery-tooling`](/delivery-tooling/) directory contains a selection of [`.go`](https://go.dev/) tooling and markdown files for use with creating a release.
+
+### Documentation
+
+The [`docs`](/docs/) directory is a centralised store for all documentation related to the project (excluding the top-level [README](/README.md) file). It contains information on the community guidelines, community policies, detailed information on governance, including the various [working groups](/docs/governance/working-groups/). The documentation directory also contains information on various reusable [resources](/docs/resources/readme.md) utilised in the project.
+
+### Schemas
+
+The [`schemas`](/schemas) directory contains the yaml validation schemas used in the project. Yaml validation schemas have been created for controls, features, metadata and threats files, to ensure that contributions follow the approved structure and contain required values.
+
+### Services
+
+The [`services`](/services/) directory is the main working directory of the project. It contains a list of common cloud services organised in a hierarchical way from high level resource type first becoming more granular. For example:
+
+```ascii
+services/
+└── database/
+    └── relational/
+    └── warehouse/
+```
+
+Each resource should contain the following files in `.yaml` format:
+
+- `controls.yaml`: The common compliance controls for the resource.
+- `features.yaml`: The common features which define the resource.
+- `threats.yaml`: The security threats related to the resource.
+- `metadata.yaml`: Metadata on the resource such as resource details.
+
+The resource should also contain a `tests` directory containing details on how to test the resource is compliant.
+
+The top level of the `services` directory contains `yaml` files for common controls, features and threats which are common to multiple resources. It also contains the [`service-families.yaml`](/services/service-families.yaml) file which documents the service families and their descriptions.
+
+## VSCode
+
+### Snippets
+
+[VSCode snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets) are defined for common [features](.vscode/common-features.code-snippets), [threats](.vscode/common-threats.code-snippets) and [controls](.vscode/common-controls.code-snippets).
+To make use of these snippets start typing the snippet prefix in any `yaml` file and VSCode will offer you the option to auto-complete the snippet e.g.
```

Comment:
Would you perhaps be able to add a grid/summary of the available snippets (general info about single common features, threats etc.).

Can be part of a different PR.

Reply:
Just to clarify, are you asking for a high-level list of each .code-snippets file and it's purpose as opposed to a list of each individual snippet and what it's for? The latter seems like a bit of duplication and would be hard to maintain.

Reply:
Just a high level list e.g.

threats-file: adds the template for a threats file
threat: adds a template for a single threat
CT*: adds a common threat

Does that make sense?

I don't see us adding many more types of snippets, other than maintaining the common features, threats and controls, but those would be covered by the general explanation.

Reply:
you mean something like [this](https://github.com/finos/common-cloud-controls/pull/541/commits/80b3253dc38bb70d7faac713eb0db96ed5585d1f)?

Reply:
Yup - just saves people having to go digging to see what is available.